sympy/matrices/determinant.py

```diff
@@ -327,6 +328,12 @@ def _adjugate(M, method="berkowitz"):
     return M.cofactor_matrix(method=method).transpose()
 
 
+def _charpoly_DOM(M, x):
+    DOM = DomainMatrix.from_Matrix(M, field=True, extension=True)
```

Comment:
Here `field=True` is not needed as the algorithm does not involve division.

Reply:
Using a field like `QQ` can be a lot slower if the corresponding ring `ZZ` can be used.

Reply:
Sure, will do, it also helps to not make any changes in test files.